src/repr/src/row.rs

```diff
@@ -461,6 +495,25 @@ fn read_byte(data: &[u8], offset: &mut usize) -> u8 {
     byte
 }
 
+fn read_byte_array_sign_extending<const N: usize>(
+    data: &[u8],
+    offset: &mut usize,
+    length: usize,
+) -> [u8; N] {
+    debug_assert!(length <= N);
+    let mut raw = [0; N];
+    if length > 0 {
+        std::io::Write::write(&mut &mut raw[0..length], &data[*offset..*offset + length]).unwrap();
+        *offset += length;
+        // This is little-endian, so the sign bit is in the last
+        // element.
+        let is_negative = (raw[length - 1] & 0x80) != 0;
+        let fill_val = if is_negative { 255 } else { 0 };
```

Comment:
Why do negatives need to fill with 255?

Reply:
That's just how two's complement integers work. For example, -96 is `0xA0` as an i8 and `0xFFFFFFFFFFFFFFA0` as an i64.